### ISSUE TITLE
Soft line break in (shift+Enter) paragrah deletes the following paragrah (Windows)

### ISSUE THREAD

**Issue by behrang1981** [WordPress/gutenberg#38322]
### Description

After moving a paragraph to end of the previous one by deleting the spaces inbetween and making a soft break with (Soft line break) the paragraph behing the curse is completely removed.

### Step-by-step reproduction instructions

1. Write two distinct paragraphs in editor
2. go the beginning of the second
3. remove the space to the first paragraph by using delete on keyboard
4. Now that both paragraphs are one make soft line break (shift + enter)
5. second part (second paragraph) is deleted

### Screenshots, screen recording, code snippet

https://i.gyazo.com/49c27b37dc0529428a9f568eecc0cedc.png

### Environment info

Windows 10
Firefox Browser latest
Gutenberg Editor
Wordpress latest (no Plugins activates)
Themes: Twenty Twenty One and Twenty Twenty Two (untouched)

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by t-hamano** [comment]
I am also a Windows user and I am suffering from this problem.

It happens not only with soft line breaks, but also with normal line breaks.


https://user-images.githubusercontent.com/54422211/151695232-a91321d2-c4e2-40f5-ad3a-652d4cdb2f88.mp4



### Environment info

- OS: Windows 11
- Browser: Firefox latest, Google Chrome latest
- WordPress version: 5.9
- Gutenberg plugin: This happens in both environments where the plugin is enabled and where it is disabled.
- Themes: Twenty Twenty One and Twenty Twenty Two

**Comment by t-hamano** [comment]
After detailed investigation, I found that **WordPress 5.7** (**gutenberg 9.9**) did not reproduce the problem, but **WordPress 5.8** (**gutenberg 10.7**) did.

So, next, on WordPress 5.7, I increased the version of the Gutenberg plugin step by step, starting with 9.9, to see which Gutenberg version caused the problem.

| Gutenberg Plugin Version | Does this symptom occur ? |
| ------------------------ | ------------------------ |
| 9.9.0                    | no                       |
| 10.5.0                   | no                       |
| 10.6.0                   | no                       |
| 10.6.2                   | no                       |
| 10.7.0                   | **yes**                  |

Therefore, I assume that the changes made between gutenberg 10.6.2 and 10.7.0 are most likely the cause of this symptom.

**Comment by behrang1981** [comment]
> After detailed investigation, I found that **WordPress 5.7** (**gutenberg 9.9**) did not reproduce the problem, but **WordPress 5.8** (**gutenberg 10.7**) did.
> 
> So, next, on WordPress 5.7, I increased the version of the Gutenberg plugin step by step, starting with 9.9, to see which Gutenberg version caused the problem.
> 
> Gutenberg Plugin Version	Does this symptom occur ?
> 9.9.0	no
> 10.5.0	no
> 10.6.0	no
> 10.6.2	no
> 10.7.0	**yes**
> Therefore, I assume that the changes made between gutenberg 10.6.2 and 10.7.0 are most likely the cause of this symptom.

Awesome job!

**Comment by t-hamano** [comment]
I found that the `container` was not being passed correctly as the first argument to the [placeCaretAtHorizontalEdge](https://github.com/WordPress/gutenberg/blob/15b28326f78fafc3a20157a1be6db06d915b74c3/packages/dom/src/dom/place-caret-at-horizontal-edge.js) function.

I debugged the following to shoot the video.

```javascript
export default function placeCaretAtHorizontalEdge( container, isReverse ) {
	console.log( container );
	return placeCaretAtEdge( container, isReverse, undefined );
}
```


https://user-images.githubusercontent.com/54422211/151700384-ab862181-9f31-4c28-8c90-509828784289.mp4



In the case where this symptom is reproduced (the problem where the second paragraph block disappears when the Enter key is pressed), the container element is empty.

But strangely enough, when I press Enter again after Undo, the correct container element comes across.

I'd like to continue to investigate, but would be happy to help.
